agents-api/agents_api/activities/task_steps/prompt_step.py

```diff
@@ -80,30 +98,65 @@ async def prompt_step(context: StepContext) -> StepOutcome:
     if isinstance(prompt, str):
         prompt = [{"role": "user", "content": prompt}]
 
-    completion_data: dict = {
-        "model": agent_model,
-        "tools": formatted_agent_tools or None,
-        "messages": prompt,
-        **agent_default_settings,
-        **passed_settings,
-    }
-
-    extra_body = {  # OpenAI python accepts extra args in extra_body
-        "cache": {"no-cache": debug},  # will not return a cached response
-    }
-
-    response = await litellm.acompletion(
-        **completion_data,
-        extra_body=extra_body,
-    )
+    # Format agent_tools for litellm
+    formatted_agent_tools = [
+        format_agent_tool(tool) for tool in agent_tools if format_agent_tool(tool)
+    ]
 
-    if context.current_step.unwrap:
-        if response.choices[0].finish_reason == "tool_calls":
-            raise ApplicationError("Tool calls cannot be unwrapped")
+    # Check if the model is Anthropic
+    if "claude-3.5-sonnet-20241022" == agent_model.lower():
+        # Retrieve the API key from the environment variable
+        betas = [COMPUTER_USE_BETA_FLAG]
+        api_key = os.getenv("ANTHROPIC_API_KEY")
```

Comment:
All the env vars should be defined in `env.py`

Reply:
kindly check once